lib/core/versions/0.11.0/TransactionProcessor.ts

```diff
@@ -222,37 +223,30 @@ export default class TransactionProcessor implements ITransactionProcessor {
     operations.push(...updateOperations);
     operations.push(...deactivateOperations);
 
-    // If chunk file is found/given, we need to add `type` and `delta` from chunk file to each operation.
-    // NOTE: there is no delta for deactivate operations.
-    const patchedOperationBuffers: Buffer[] = [];
-    if (chunkFile !== undefined) {
-
-      // TODO: Issue 442 - https://github.com/decentralized-identity/sidetree/issues/442
-      // Use actual operation request object instead of buffer.
-
-      const operationCountExcludingDeactivates = createOperations.length + recoverOperations.length + updateOperations.length;
-      for (let i = 0; i < operationCountExcludingDeactivates &&
-                      i < chunkFile.deltas.length; i++) {
-        const operation = operations[i];
-        const operationJsonString = operation.operationBuffer.toString();
-        const operationObject = await JsonAsync.parse(operationJsonString);
-        operationObject.type = operation.type;
-        operationObject.delta = chunkFile.deltas[i];
-
-        const patchedOperationBuffer = Buffer.from(JSON.stringify(operationObject));
-        patchedOperationBuffers.push(patchedOperationBuffer);
-      }
-    }
+    // TODO: Issue 442 - https://github.com/decentralized-identity/sidetree/issues/442
+    // Use actual operation request object instead of buffer.
 
-    // Add anchored timestamp to each operation.
+    // NOTE: The last set of `operations` are deactivates, they don't have `delta` property.
     const anchoredOperationModels = [];
     for (let i = 0; i < operations.length; i++) {
       const operation = operations[i];
+      const operationJsonString = operation.operationBuffer.toString();
+      const operationObject = await JsonAsync.parse(operationJsonString);
+      operationObject.type = operation.type;
+
+      // Add `delta` property read from chunk file to each operation if chunk file exists.
+      if (operation.type === OperationType.Deactivate) {
```

Comment:
nit: can't we just use 1 if statement, the comments should explain the logic well enough that we don't need an empty if block and then else if